packages/service-provider-server/src/cli-service-provider.ts

```diff
@@ -1092,10 +1098,15 @@ class CliServiceProvider extends ServiceProviderCore implements ServiceProvider
       authMechanism: authDoc.mechanism,
       authSource: authDoc.authDb
     };
-    this.mongoClient = await MongoClient.connect(
```

Comment:
This was a bug where we were not closing old MongoClients upon reauthentication.